### PR TITLE
Add repo documentation

### PR DESCRIPTION
**Pull request by rmfan**
Adds Readme's to `src/graph` and `src/graph_service`

### CODE REVIEW

File: src/graph/README.md [hunk 1]
```diff
@@ -1,9 +1,206 @@
+# Rampart Controller
+
+## Overview
+
+The Rampart Controller serves as the [Kubernetes Controller](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) for the RampartGraph custom resource. It contains three seperate components with the following rolls:
```
Comment (jingjiezhu) [inline src/graph/README.md]: Is this url correct?

File: src/graph/README.md [hunk 2]
```diff
@@ -1,9 +1,206 @@
...
+Within the code itself, phase 2 is performed during instatiation of the `Graph` object by the classmethod `from_json`. Phases 3 and 4 are performed by the `validate` function, with phase 4 only performed when the argument `validate_specs` is set to `True`.
+
+### Deployment
+
+Once validation is completed, the controller can begin deployment of components and flows. To do this, it performs the following steps:
+
+1. Deploy all of the subnamespaces that will contain the components (note: each component gets its own namespace)
+2. Deploy all of the flows. Note that the actual body of volume flows do not exist per se, but are rather when two components using it write and read to the same place on a distributed filesystem
+   a. Create Pulsar tenants, namespaces, and topics to house Pulsar flows
+   b. Create all the auxiliary volume claimsm, deployments, and services to monitor volume flows
```
Comment (jingjiezhu) [inline src/graph/README.md]: typo: `claimsm` => `claims`

File: src/graph/README.md [hunk 3]
```diff
@@ -1,9 +1,206 @@
...
+
+Within the code itself, phase 2 is performed during instatiation of the `Graph` object by the classmethod `from_json`. Phases 3 and 4 are performed by the `validate` function, with phase 4 only performed when the argument `validate_specs` is set to `True`.
+
+### Deployment
+
+Once validation is completed, the controller can begin deployment of components and flows. To do this, it performs the following steps:
+
+1. Deploy all of the subnamespaces that will contain the components (note: each component gets its own namespace)
+2. Deploy all of the flows. Note that the actual body of volume flows do not exist per se, but are rather when two components using it write and read to the same place on a distributed filesystem
```
Comment (jingjiezhu) [inline src/graph/README.md:33]: In view mode, there is no line break after this sentence.
https://github.com/petuum/rampart/blob/0f1a29aaaa0c4aced83a3dbfdf0947c609cd9ad8/src/graph/README.md#deployment

File: src/graph/README.md (hunk 1 above)
Comment (rmfan) [reply src/graph/README.md]: Yes, but ill reword the sentence to make it fit better

**Review by jingjiezhu — APPROVED** [review]
LGTM

File: src/graph/README.md [hunk 4]
```diff
@@ -1,9 +1,206 @@
+# Rampart Controller
+
+## Overview
+
+The Rampart Controller serves as the controller of the [Kubernetes Operator](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) for the RampartGraph custom resource. It contains three seperate components with the following rolls:
```
Comment (DianLiI) [inline src/graph/README.md]: roles?

File: src/graph_service/README.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
+# Rampart Backend Service
+
+## Overview
+
+The Rampart Backend Service provides a backend for the Rampart UI to interact with Kubernetes. The Rampart UI has the concept of `App`s, which are Rampart graphs with version history (including with the ability to revert, etc). Unfortunately, this abstraction does not match the abstraction maintained by `RampartGraph` custom resources, which do not maintain any sort of versioning. Therefore, the Rampart Backend Service needs to store the mapping between `App`s and `RampartGraph` custom resources.
```
Comment (DianLiI) [inline src/graph_service/README.md]: Rampart UI does not interact with Kubernetes directly. I think it is better to say something like "... backend for Rampart UI to Interactively build rampart graphs and manage graph deployments."

Comment (rmfan) [reply src/graph_service/README.md]: Updated

**Comment by DianLiI** [comment]
Maybe add links (relative file path would work) to the new docs in the root README file.

**Review by jingjiezhu — APPROVED** [review]

**Review by DianLiI — APPROVED** [review]

**Pull request merged**